src/AutoUI/Lenses/types/table.tsx

```diff
@@ -2,13 +2,167 @@ import React from 'react';
 import { faTable } from '@fortawesome/free-solid-svg-icons/faTable';
 import type { LensTemplate } from '..';
 import type { CollectionLensRendererProps } from '.';
-import type { TableColumn } from 'rendition';
-import { Table } from 'rendition';
-import { setToLocalStorage } from '../../utils';
-import pick from 'lodash/pick';
-import { useAnalyticsContext } from '@balena/ui-shared-components';
+import {
+	DEFAULT_ITEMS_PER_PAGE,
+	getFromLocalStorage,
+	setToLocalStorage,
+} from '../../utils';
+import { Table } from '../../../components/Table';
+import type { AutoUIEntityPropertyDefinition } from '~/AutoUI';
+import type {
+	TableSortOptions,
+	TagField,
+} from '../../../components/Table/utils';
+import { useColumns } from '../../../components/Table/useColumns';
+import { useTagActions } from '../../../components/Table/useTagActions';
+import { AddTagHandler } from '../../../components/Table/AddTagHandler';
 
-const TableColumnStateStoredProps = ['key', 'selected', 'type', 'tagKey'];
+import {
+	designTokens,
+	Material,
+	Copy,
+	useAnalyticsContext,
+} from '@balena/ui-shared-components';
+const { Box, styled, Tooltip, Typography } = Material;
+
+const TAG_COLUMN_PREFIX = 'tag_column_';
+
+const TagContainer = styled(Box)(() => ({
+	border: `0.5px solid ${designTokens.color.border.accent.value}`,
+	borderRadius: '3px',
+	color: designTokens.color.text.value,
+	backgroundColor: designTokens.color.bg.accent.value,
+	position: 'relative',
+	width: 'fit-content',
+}));
+
+export interface TagLabelProps {
+	value: string;
+}
+
+export const TagLabel = ({ value }: TagLabelProps) => {
+	return (
+		<Tooltip title={value}>
+			<TagContainer>
+				<Copy copy={value} variant="absolute">
+					<Box display="flex" gap={1} mx={2} my={1}>
+						<Typography
+							variant="bodySm"
+							noWrap
```

Comment:
Have you confirmed the `noWrap` works with the parent container being display flex? If it does not actually work, you may need to change either the Box or the Typography itself to display `block` or `inline-block`

Reply:
yes sir, it works as expected